electrum_nmc/electrum/names.py

```diff
@@ -132,6 +132,30 @@ def validate_value_length(value: bytes):
     if value_length > value_length_limit:
         raise BitcoinException('value length {} exceeds limit of {}'.format(value_length, value_length_limit))
 
+def validate_i2p_address(address: str) -> None:
+    try:
+        # Check length
+        if len(address) > 67:
+            raise ValueError("Address length exceeds 67 characters")
+
+        # Check if it ends with '.b32.i2p'
+        if address.endswith('.b32.i2p'):
```

Comment:
Inverting this if condition, so that the exception shows up above the later checks, would make the code more readable.